README.md

```diff
@@ -51,6 +51,41 @@ Login with one of these users:
 - Username `purchaser` password `password`
 - Username `approver` password `password`
 
+## Using OAuth 2.0 Authentication
+This application supports both Http Basic authentication and OAuth 2.0 authentication.
+To use OAuth 2.0 authentication, you will need to set up a Keycloak server as the OAuth 2.0 provider.
+
+### Using Keycloak server
+1. Download, install and start a Keycloak server (19.0.0 or later) with command line.
+    ```
+    bin\kc.[sh|bat] start-dev --http-port=8081
+    ```
+2. Go to Keycloak admin console(e.g. http://localhost:8081) to make sure the service has started successfully. Create an initial admin user.
+3. Download [demo-app-realm.json](https://github.com/parasoft/parasoft-demo-app/blob/main/keycloak/demo-app-realm.json).
+    >  [demo-app-realm.json]() includes a realm *demo-app-realm* and necessary information to use Keycloak with demo application:
+    > * 50 purchasers and 50 approvers which have consistent credentials and roles with default data in database of demo application.
+    > * A client with `demo-app-client` as id and `DzOS5Y4iRmHIQH6ntTGHj78PpFEjUKLo` as secret.
+    > * Valid redirect URI set as `*`, which allows redirection to any URI.
+
+4. Run command line in root directory of Keycloak to import predefined realm data.
+    ```
+    bin/kc.[sh|bat] import --file path/to/demo-app-realm.json
+    ```
+5. If Keycloak server is not running on *localhost:8081*, make sure to change `spring.security.oauth2.client.provider.keycloak.realm-rui` property value in **application.properties** file.
+6. Start the demo application and choose to log in with OAuth 2.0.
+
+### Using SOAtest OAuth 2.0 Authentication (Authorization Code type)
+
+**Configuration details for Authorization Code type (default)**
```

Comment:
I think we can also add "Property" column here and add these properties to the example application.properties that Renee added. It's still possible that user could use a Keycloak server with different configuration.